src/Utils/QueryAnalyzer.php

```diff
@@ -531,16 +556,95 @@ public function get_headers( array $headers = [] ): array {
 			// Iterate over the chunks
 			foreach ( $chunks as $index => $chunk ) {
 				if ( 0 === $index ) {
-					$headers['X-GraphQL-Keys']      = $chunk;
-					$headers['X-GraphQL-Keys-Size'] = strlen( $chunk );
+					$return_keys = $chunk;
 				} else {
-					$headers[ 'X-GraphQL-Skipped-Keys-' . $index ]           = $chunk;
-					$headers[ 'X-GraphQL-Skipped-Keys-' . $index . '-Size' ] = strlen( $chunk );
+					$this->skipped_keys = trim( $this->skipped_keys . ' ' . $chunk );
 				}
 			}
 		}
 
+		$skipped_keys_array = ! empty( $this->skipped_keys ) ? explode( ' ', $this->skipped_keys ) : '';
+
+		$this->graphql_keys = [
+			'keys'             => $return_keys,
+			'keysLength'       => strlen( $return_keys ),
+			'skippedKeys'      => $this->skipped_keys,
+			'skippedKeysCount' => ! empty( $skipped_keys_array ) ? count( $skipped_keys_array ) : 0,
+		];
+
+		return $this->graphql_keys;
+
+	}
+
+
+	/**
+	 * Return headers
+	 *
+	 * @param array $headers The array of headers being returned
+	 *
+	 * @return array
+	 */
+	public function get_headers( array $headers = [] ): array {
+
+		$keys = $this->get_graphql_keys();
+
+		if ( ! empty( $keys ) ) {
+
+			$headers['X-GraphQL-Query-ID']  = $this->query_id ?: null;
+			$headers['X-GraphQL-Keys']      = $keys['keys'] ?: null;
+			$headers['X-GraphQL-Keys-Size'] = strlen( $keys['keys'] ) ?: 0;
+			if ( 0 < absint( $keys['skippedKeysCount'] ) ) {
+				$headers['X-GraphQL-Skipped-Keys-Count'] = $keys['skippedKeysCount'] ?: 0;
```

Comment:
I might quibble with using `Size` for the total and `Count` for the skip count. 

Maybe use count for both?
`X-GraphQL-Keys-Count`
`X-GraphQL-Skipped-Keys-Count`

Reply:
If it's only useful for debugging and we're worried about header size, maybe move this kind of info to the response body `extensions`.

I think we should limit response headers to things that _need_ to be present for systems interconnections, e.g., Varnish, etc.